src/main/java/Duke.java

```diff
@@ -1,10 +1,259 @@
+        FileWriter fw = new FileWriter(filePath, true);
+        fw.write(taskToAdd.toData());
+        fw.write("\n");
+        fw.close();
+    }
+
+    private static ArrayList<Task> loadData(File file, ArrayList<Task> list) throws DukeException{
+        try {
+            Scanner s = new Scanner(file);
+            while (s.hasNextLine()) {
+                String line = s.nextLine();
+                String taskType = line.substring(0, 1);
+                switch (taskType) {
+                case "T":
+                    list.add(Todo.dataToTask(line.substring(4)));
+                    break;
+                case "E":
+                    list.add(Event.dataToTask(line.substring(4)));
+                    break;
+                case "D":
+                    list.add(Deadline.dataToTask(line.substring(4)));
+                    break;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
+        return list;
+    }
+
+    private static void writeAllToFile(ArrayList<Task> list, File f) {
+        try {
+            FileWriter fw = new FileWriter(f);
+            for (int i = 0; i < list.size(); i++) {
+                fw.write(list.get(i).toData());
+                fw.write("\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+    public static void main(String[] args) throws DukeException {
+        String line = "──────────────────────────────────────────────────────────────────────────";
+        String logo = " _____   __                 _____ _           _   _           _  ___\n"
+                + "|  _\\ \\ / /                /  __ \\ |         | | | |         | ||_  |\n"
+                + "| |  \\ V /___  _   _ _ __  | /  \\/ |__   __ _| |_| |__   ___ | |_ | |\n"
+                + "| |   \\ // _ \\| | | | '__| | |   | '_ \\ / _` | __| '_ \\ / _ \\| __|| |\n"
+                + "| |   | | (_) | |_| | |    | \\__/\\ | | | (_| | |_| |_) | (_) | |_ | |\n"
+                + "| |_  \\_/\\___/ \\__,_|_|     \\____/_| |_|\\__,_|\\__|_.__/ \\___/ \\__|| |\n"
+                + "|___|                                                           |___|\n";
+
+        String greet = line
+                + "\n"
+                + "Hello! I'm\n"
+                + logo
+                + "What can I do for you?\n"
+                + line
+                + "\n";
+        System.out.println(greet);
+        Scanner input = new Scanner(System.in);
+
+        ArrayList<Task> list = new ArrayList<>();
+        String filepath = "data/duke.txt";
+        makeDataDir();
+        File f = new File(filepath);
+        if (!createFile(f)) {
+            try {
+                loadData(f, list);
+            } catch (DukeException e) {
+                throw e;
+            }
+        }
+
+        boolean end = false;
+        while (!end) {
```

Comment:
Perhaps consider abstracting your if else or switch case statements so there's not much indentation here?

Reply:
Thanks for the feedback, I've taken that into consideration and updated my code with more abstraction.